Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  SignupController.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/7/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+public final class SignupController: UIViewController {
+    
+    private let _viewModel: SignupViewModelType
+    private let _signupViewFactory: () -> SignupViewType
+    private let _delegate: SignupControllerDelegate
+    
+    public lazy var signupView: SignupViewType = self._signupViewFactory()
+    
+    private lazy var _notificationCenter: NSNotificationCenter = NSNotificationCenter.defaultCenter()
+    private var _disposable = CompositeDisposable()
+    private let _keyboardDisplayed = MutableProperty(false)
+    private let _activeTextField = MutableProperty<UITextField?>(.None)
+
+    
+    init(viewModel: SignupViewModelType, signupViewFactory: () -> SignupViewType, delegate: SignupControllerDelegate = DefaultSignupControllerDelegate()) {
```

Comment:
There are 2 things here. The first one is that I think that we should use `internal` explicitly everywhere. Maybe for our internal project we should discuss it with the rest of the team but the OSS projects I would follow the GitHub guidelines quite close. I think it makes everything way more clear to explicitly state that this method or class is for internal use only.

In this particular case I agree with @danielaRiesgo. The initializer should be `internal` because the controller is not intended for customization because is just glue code between the view and the view model. Unlike the view or view model where there are protocols and if you want to have a different implementation of the view for example is just a matter of creating a new class and conforming to the protocol.